tangelo/toolboxes/circuits/lcu.py

```diff
@@ -20,6 +20,7 @@
 
 import math
 from typing import Union, Tuple, List
+from copy import copy
```

Comment:
To make a shallow copy of a list just use the built-in copy method or slicing.
If your control lists contain integers (e.g immutable values) then the shallow copy is equivalent to a deepcopy.

```
l = [1,2,3]
l1 = l[:] # Full copy, via slicing
l2 = l.copy() # Shallow copy. Ok with integers etc

l.append('a')
l1.append('b')
l2.append('c')

# Each list is indeed separate / different
print(l, l1, l2)
```